File: increasing-triplet-subsequence/increasing-triplet-subsequence.py
class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        # if length of input is 2 or less, return False
        if len(nums) <= 2:
            return False
        # if there are only 2 or less unique numbers in the array, return False
        elif len(set(nums)) <= 2:
            return False
        else:
            # keep track of lowest number found as we iterate through array
            minNum = None

            for i in range(0, len(nums)-1):
                currentNum = nums[i]

                # if the current number is less than minNum found or no minNum exists
                if (currentNum < minNum) or (minNum is None):
                    minNum = currentNum
                # if the current number is greater than minNum found previously
                elif (currentNum > minNum):
                    for j in range(i+1, len(nums)):
                        if nums[j] > currentNum:
                            return True
                # if the current number is equal to minNum
                else:
                    continue

            return False

File: increasing-triplet-subsequence/increasing-triplet-subsequence.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */

// RUN TIME IS SLOWER COMPARED TO OTHERS - ONLY BEATS 9.89% of other users

var increasingTriplet = function(nums) {
    if (nums.length <= 2) {
        return false
    } else if (new Set(nums).size <= 2) {
        return false
    } else {
        var minNum = Infinity
        for (let i=0; i < nums.length-1; i++) {
            var currentNum = nums[i]
            if (currentNum < minNum) {
                minNum = currentNum
            } else if (currentNum > minNum) {
                for (let j=i+1; j < nums.length; j++) {
                    if (nums[j] > currentNum) {
                        return true
                    }
                }
            } else {
                continue
            }
        }
        return false
    }
};